fpdf/fpdf.py

```diff
@@ -2798,6 +2797,32 @@ def local_context(
                 "cannot create a local context inside an unbreakable() code block"
             )
         self._push_local_stack()
+        self._init_local_context(**kwargs)  # write "q" in the output stream
+        yield
+        self._out("Q")
+        self._pop_local_stack()
+
+    def _init_local_context(
```

Comment:
Maybe it would make sense to have an `_exit_local_context()` as well?
It would be very simple, but make the code less confusing (no searching for the implicit `"q"` matching the explicit `"Q"`).